bika/lims/subscribers/bikasetup.py

```diff
@@ -38,10 +38,33 @@ def BikaSetupModifiedEventHandler(instance, event):
     mp(AddClient, roles, 1)
     mp(EditClient, roles, 1)
     # Set permissions at object level
-    # TODO-performance: We are allways reindexing all clients
-    for obj in portal.clients.objectValues():
-        mp = obj.manage_permission
-        mp(AddClient, roles, 0)
-        mp(EditClient, roles, 0)
-        mp(permissions.ModifyPortalContent, roles, 0)
-        obj.reindexObject()
+    clients = portal.clients.objectValues()
+    if check_if_client_permissions_has_changed(clients[0], roles):
+        for obj in clients:
+            mp = obj.manage_permission
+            mp(AddClient, roles, 0)
+            mp(EditClient, roles, 0)
+            mp(permissions.ModifyPortalContent, roles, 0)
+            obj.reindexObject()
+
+
+def check_if_client_permissions_has_changed(client, roles):
```

Comment:
I think the function `check_if_client_permissions_has_changed` could have a better name, like `client_permissions_changed`